src/engraving/dom/chordlist.cpp

```diff
@@ -1685,6 +1685,10 @@ void ChordList::read(XmlReader& e)
                     String code = e.attribute("code");
                     String symClass = e.attribute("class");
                     if (!code.empty()) {
+                        if (!code.startsWith(u"0x") && !code.startsWith(u"0") && !code.startsWith('&') && !code.endsWith(';')) {
+                            // fix broken chord lists, see https://github.com/musescore/MuseScore/issues/16363
+                            code = u"0x" + code;
+                        }
                         bool ok = true;
                         char32_t val = code.toUInt(&ok, 0);
```

Comment:
If `code.startsWith('&')` is true, then this would fail anyway. Does this case occur at all?
If not, let's not check for it, because that only makes the code noisy
If yes, we should add proper handling for that case.

However, I'm not very confident whether I like this approach; it means that our `String::number` method behaves differently from `QString::number`, which seems a potential source of confusion. 
Wouldn't the problem be fixed instantly if you do 
```suggestion
                        char32_t val = code.toUInt(&ok, 16);
```
and drop all other changes? Or is it not certain that it will be a base16 number?

Reply:
> If `code.startsWith('&')` is true, then this would fail anyway. Does this case occur at all? If not, let's not check for it, because that only makes the code noisy If yes, we should add proper handling for that case.

We don't have full control over the content of a chords.xml, it could contain "&flat;" or "&#9837;" (@MarcSabatella even recommeded the latter in the issue as a workaround, https://github.com/musescore/MuseScore/issues/16363#issuecomment-1430235880).

The toInt() would fail on it, but return the string unchanged, as far as I understand.

Be generous in what you accept, but strict in what you generate ;-)
 
> However, I'm not very confident whether I like this approach; it means that our `String::number` method behaves differently from `QString::number`, which seems a potential source of confusion. Wouldn't the problem be fixed instantly if you do
> ... 
> and drop all other changes? Or is it not certain that it will be a base16 number?

The problem isn't that it doesn't return a hex number, but that this number as a string doesn't look like one to the writer because of the missing "0x".

Reply:
But we can leave the String::number() as it was and just do t like in the 1st commit and prepend the "0x" there

Reply:
We know though that we **write** as hex

Reply:
I can understand your concerns reg. different behavoir of `String::number()` vs. `QString::number()`, so reverted that 2nd commit

Reply:
And we may not need the read fix at all, as  save/close/open fixes the issue too, its just nice to have, isn't it?

Reply:
BTW: this would work too:
```cpp
                        int base = 0; // guess, based on content
                        if (!code.startsWith(u"0x") && !code.startsWith(u"0") && !code.startsWith('&') && !code.endsWith(';')) {
                            // fix broken chord lists, see https://github.com/musescore/MuseScore/issues/16363
                            base = 16; // assume hex
                        }
                        bool ok = true;
                        char32_t val = code.toUInt(&ok, base);
```
And it does read that `&#9837;` too